packages/flutter_tools/templates/list_detail_app/lib/src/settings/settings_controller.dart.tmpl

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import 'settings_service.dart';
+
+/// A class that many Widgets can interact with to read user settings, update
+/// user settings, or listen to user settings changes.
+///
+/// Controllers glue Data Services to Flutter Widgets. The SettingsController
+/// uses the SettingsService to store and retrieve user settings.
+class SettingsController with ChangeNotifier {
+  SettingsController(this._settingsService);
+
+  // Make SettingsService a private variable so it is not used directly.
+  final SettingsService _settingsService;
+
+  // Make ThemeMode a private variable so it is not updated directly without
+  // also persisting the changes with the SettingsService.
+  ThemeMode _themeMode;
+
+  // Allow Widgets to read the user's preferred ThemeMode.
+  ThemeMode get themeMode => _themeMode;
+
+  /// Load the user's settings from the SettingsService. It may load from a
+  /// local database or the internet. The controller only knows it can load the
+  /// settings from the service.
+  Future<void> loadSettings() async {
+    _themeMode = await _settingsService.themeMode();
+
+    // Important! Inform listeners a change has occurred.
+    notifyListeners();
+  }
+
+  /// Update and persist the ThemeMode based on the user's selection.
+  Future<void> updateThemeMode(ThemeMode newTheme) async {
+    _themeMode = newTheme;
```

Comment:
technically, we only need to persist and inform if `newTheme != _themeMode`?

Reply:
For sure. I wasn't sure if it made sense to add performance optimizations in at this point, but we could return early from this method if the `newTheme` is equal to the old theme. What do you think?

Reply:
I think I would include this here with a quick comment explaining the rational.

Reply:
Sure, no problem. I've added it in!